CollectAudioLogs/CollectAudioLogs.ps1

```diff
@@ -1,4 +1,87 @@
 ﻿# Copyright (c) Microsoft Corporation. All rights reserved.
+param(
+    [string]$problem,
+    [switch]$reproLogsOnly
+);
+
+Function Get-YesNoResponse
+{
+    param([string]$prompt);
+    do 
+    {
+        $response = Read-Host "${prompt}? (y/n)";
+        $response = $response.Trim().ToLower();
+    }
+    while (-not $response.StartsWith("y") -and -not $response.StartsWith("n"));
+    
+    Return $response.StartsWith("y");
+}
+
+[Nullable[bool]]$canRepro;
+[Nullable[bool]]$glitch;
+[bool]$ttt = $false;
+
+switch ($problem)
+{
+    ""
+    {
+    }
+
+    "glitch"
+    {
+        $canRepro = $true;
+        $glitch = $true;
+        break;
+    }
+
+    "norepro"
+    {
+        $canRepro = $false;
+        break;
+    }
+
+    "other"
+    {
+        $canRepro = $true;
+        $glitch = $false;
+        break;
+    }
+
+    "ttt"
+    {
+        $canRepro = $true;
+        $glitch = $false;
+        $ttt = $true;
+        break;
+    }
+
+    default
+    {
+        Write-Host "Unexpected problem $problem - specify 'glitch', 'norepro', or 'other'";
+        Exit;
+    }
+}
+
+if ($null -eq $canRepro)
+{
+    # Ask the user if they'd like to take a repro trace
+    $canRepro = Get-YesNoResponse -prompt "Do you want to grab logs of the problem in action";
+}
+
+if ($canRepro)
+{
+    if ($null -eq $glitch)
+    {
+        # Ask the user if the problem is an audio glitch
+        $glitch = Get-YesNoResponse -prompt "Is the problem that audio sounds bad";

```

Comment:
I like this, it's a clear way to determine what sort of log to grab.

I think if it's not a glitch log, then we should adjust what is logged. The glitch logging adds a massive amount of trace, which reduces the amount of time that the log can collect to around 30 seconds. We regularly have repro logs where the user who collected the log collects more than 30 seconds worth of steps and have audio playing, so by the time they stop the log the interesting bits are gone.

So, i think it would be beneficial to either give an option to the user for >30 seconds worth of logging, even though the logs will be larger, or modify the non-glitch logs to remove tracing for individual buffers/glitch information.

Reply:
Looking closer at the new audio-info.wprp, i think it's probably already good, since the new wprp no longer has all of the cpu and dpc info. I'll have to see how it actually performs, but hopefully it'll hold more than 30 seconds worth of logging when playback is active. :)

Reply:
The goal for glitch logging is to have around a minute of effective lookback, and the goal for "other" logging is to have a much longer lookback (tens of minutes at least)